apps/interpreter/src/run-options.ts

```diff
@@ -0,0 +1,181 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import {
+  type DebugGranularity,
+  DebugGranularityValues,
+  type DebugTargets,
+  DefaultDebugTargetsValue,
+  type Logger,
+  isDebugGranularity,
+} from '@jvalue/jayvee-execution';
+
+export interface RunOptions {
+  pipeline: string;
+  env: Map<string, string>;
+  debug: boolean;
+  debugGranularity: DebugGranularity;
+  debugTarget: DebugTargets;
+  parseOnly: boolean;
+}
+
+export function parseRunOptions(
+  optionsRaw: unknown,
+  logger: Logger,
+): RunOptions | undefined {
+  if (typeof optionsRaw !== 'object' || optionsRaw == null) {
+    logger.logErr(
+      "Error in the interpreter: didn't receive a valid RunOptions object.",
+    );
+    return undefined;
+  }
+
+  const requiredFields = [
+    'pipeline',
+    'env',
+    'debug',
+    'debugGranularity',
+    'debugTarget',
+    'parseOnly',
+  ];
+  if (requiredFields.some((f) => !(f in optionsRaw))) {
+    logger.logErr(
+      `Error in the interpreter: didn't receive a valid RunOptions object. Must have the fields ${requiredFields
+        .map((f) => `"${f}"`)
+        .join(', ')} but got object ${JSON.stringify(optionsRaw, null, 2)}`,
+    );
+    return undefined;
+  }
+
+  const options = optionsRaw as {
+    pipeline: unknown;
+    env: unknown;
+    debug: unknown;
+    debugGranularity: unknown;
+    debugTarget: unknown;
+    parseOnly: unknown;
+  };
```

Comment:
```suggestion
  const options = optionsRaw as Record<keyof RunOptions, unknown>;
```

This way we don't require any changes here when we change `RunOptions`.